react/src/components/ImageInstallModal.tsx

```diff
@@ -0,0 +1,197 @@
+        imageResource.fgpu = imageResource['cuda.shares'];
+      } else if ('cuda.device' in imageResource) {
+        imageResource.gpu = imageResource['cuda.device'];
+        isGPURequired = true;
+      } else {
+        isGPURequired = false;
+      }
+
+      // Add 256m to run the image.
+      if (imageResource.mem.endsWith('g')) {
+        imageResource.mem = imageResource.mem.replace('g', '.5g');
+      } else if (imageResource.mem.endsWith('m')) {
+        imageResource.mem = Number(imageResource.mem.slice(0, -1)) + 256 + 'm';
+      }
+
+      imageResource.domain = baiClient._config.domainName;
+      imageResource['group_name'] = baiClient.current_group;
+
+      const resourceSlots = await baiClient.get_resource_slots();
+
+      if (isGPURequired) {
+        if (
+          !('cuda.device' in resourceSlots) &&
+          !('cuda.shares' in resourceSlots)
+        ) {
+          delete imageResource['gpu'];
+          delete imageResource['fgpu'];
+          delete imageResource['cuda.shares'];
+          delete imageResource['cuda.device'];
+        }
+      }
+
+      if ('cuda.device' in resourceSlots && 'cuda.shares' in resourceSlots) {
+        // Can be possible after 20.03
+        if ('fgpu' in imageResource && 'gpu' in imageResource) {
+          delete imageResource['gpu'];
+          delete imageResource['cuda.device'];
+        }
+      } else if ('cuda.device' in resourceSlots) {
+        delete imageResource['fgpu'];
+        delete imageResource['cuda.shares'];
+      } else if ('cuda.shares' in resourceSlots) {
+        delete imageResource['gpu'];
+        delete imageResource['cuda.device'];
+      }
+
+      imageResource.enqueueOnly = true;
+      imageResource.type = 'batch';
+      imageResource.startupCommand = 'echo "Image is installed"';
+
+      upsertNotification({
+        message: `${t('environment.InstallingImage')}${imageName}${t('environment.TakesTime')}`,
+        open: true,
+        duration: 2,
+      });
+
+      //@ts-ignore
+      const indicator = await globalThis.lablupIndicator.start('indeterminate');
+      indicator.set(10, t('import.Downloading'));
+      baiClient.image
+        .install(imageName, image?.architecture, imageResource)
+        .then(() => {
+          indicator.end(1000);
+        })
+        .catch((error: any) => {
+          setInstallingImages(
+            imagesToInstall
+              .map((item) => item?.id)
+              .filter(
+                (id): id is string =>
+                  id !== null && id !== undefined && id !== image?.id,
+              ),
+          );
+          // @ts-ignore
+          globalThis.lablupNotification.text = painKiller.relieve(error.title);
+          // @ts-ignore
+          globalThis.lablupNotification.detail = error.message;
+          // @ts-ignore
+          globalThis.lablupNotification.show(true, error);
+          indicator.set(100, t('environment.DescProblemOccurred'));
+          indicator.end(1000);
```

Comment:
Is there a reason why you didn't use `upsertNotification` in this part?

Reply:
- The way to display the error in upsertNotification seemed to be to set the backgroundTask status to rejected.
But there is no backgroundTask.
- I also referenced the container Registry List code 
ref) https://github.com/lablup/backend.ai-webui/blob/main/react/src/components/ContainerRegistryList.tsx#L159

Reply:
@gahyuun  we can change this part using `upsertNotification` with `promise`. This is an [example](https://github.com/lablup/backend.ai-webui/blob/81c1dacb35a17b74da5c1e053097f511fd9e382d/react/src/pages/SessionLauncherPage.tsx#L502-L515).